tests/unit/test_environment.py

```
@@ -0,0 +1,181 @@
+    def test_tty_but_agent_is_not_interactive(self, clean_env):
+        clean_env.setenv("CI", "1")
+        with mock.patch("sys.stdout.isatty", return_value=True):
+            assert is_interactive_terminal() is False
+
+    def test_tty_but_no_color_is_not_interactive(self, clean_env):
+        clean_env.setenv("NO_COLOR", "1")
+        with mock.patch("sys.stdout.isatty", return_value=True):
+            assert is_interactive_terminal() is False
+
+
+def _ns(**kw):
```

Ruff (0.15.15)

[warning] Missing return type annotation for private function `_ns`

(ANN202)